### PR TITLE
Create button interactor

### PR DESCRIPTION
**Pull request by elrickvm**
## Motivation
Button are the primary mechanism or utility for users of an Application to interact with the User interface.

## Approach
Create Button interactor that allows Developers to test for user interaction with button in their test cases

### CODE REVIEW

File: test/button.test.tsx [hunk 1]
```diff
@@ -1,9 +1,211 @@
...
-  .assertion(btButton("CLICK ME").exists());
+  .child("rendering a button", (test) => test
+    .step(render("render button", <MuiButton>My Button</MuiButton>))
+    .assertion(button.exists())
+    .assertion(button.has({ text: "My Button".toUpperCase()}))
+
+    .assertion("should render with the root, text and textPrimary classes but no others", async() => {
+    await button.has({ classList: containing(["MuiButton-root", "MuiButton-text"]) })
+    await button.has({ classList: not(containing(["MuiButton-textSecondary", "MuiButton-outlined", "MuiButton-contained"])) })
+    })
```
Comment (cowboyd) [inline test/button.test.tsx]: We can make these individual assertions, right?

**Review by cowboyd — COMMENTED** [review]
Only have one comment on the tests, otherwise, looks like we need a changeset and go?

File: test/button.test.tsx [hunk 2]
```diff
@@ -1,9 +1,211 @@
...
+  .child('renders button with endIcon', (test) => test
+    .step(render(<MuiButton endIcon={<span>icon</span>}>EndIcon Button</MuiButton>))
+    .assertion(button.has({ className: including("MuiButton-root") }))
+    .assertion(button.has({ className: including("MuiButton-text") }))
+    .assertion(button.has({ iconLabel: including("MuiButton-endIcon") }))
+    .assertion(button.has({ iconLabel: not(including("MuiButton-startIcon")) }))
+  )
+  .child('button renders with ripple by default', (test) => test
+    .step(render(<MuiButton TouchRippleProps={{ className: 'touch-ripple'}}>Ripple Button</MuiButton>))
+    .assertion(button.find(HTML({ className: including('touch-ripple')})).exists())
```
Comment (elrickvm) [inline test/button.test.tsx:159]: @cowboyd line 158 is failing with:

```js
Argument of type 'Interactor<HTMLElement, MergeObjects<Record<never, never>, { text?: string | NewMatcher<string> | OldMatcher<string> | undefined; title?: string | NewMatcher<...> | OldMatcher<...> | undefined; ... 4 more ...; focused?: boolean | ... 2 more ... | undefined; }>> & ActionMethods<...> & Omit<...>' is not assignable to parameter of type 'Interactor<any, any>'.ts(2345)
```

Any ideas?

**Comment by elrickvm** [comment]
Thank you @wKich!!!

**Review by cowboyd — APPROVED** [review]
Looks good on my end

**Pull request merged**